articles/understanding-loss-functions-in-machine-learning/index.md

```diff
@@ -0,0 +1,101 @@
+
+[Huber Loss](https://en.wikipedia.org/wiki/Huber_loss) combines the robustness of L1 with the stability of L2, essentially the best of L1 and L2 losses. For huge errors, it is linear and for small errors, it is quadratic in nature. Huber Loss is characterized by the parameter delta (𝛿).
+
+![Huber Loss](/engineering-education/understanding-loss-functions-in-machine-learning/huber-loss.PNG)
+
+*Source: [Analytics Vidhya](https://cdn.analyticsvidhya.com/wp-content/uploads/2019/08/huber.jpg.jpg)*
+
+### Loss functions for classification
+Classification problems involve predicting a discrete class output. It involves dividing the dataset into different and unique classes based on different parameters so that a new and unseen record can be put into one of the classes. A mail can be classified as a spam or not a spam and a person's dietary preferences can be put in one of three categories - vegetarian, non-vegetarian and vegan. Let's take a look at loss functions that can be used for classification problems.
+
+#### Binary Cross Entropy Loss
+This is the most common loss function used for classification problems that have two classes. The word "entropy", seemingly out-of-place, has a statistical interpretation. [Entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) is the measure of randomness in the information being processed. If the divergence of the predicted probability from the actual label increases, the cross-entropy loss increases. Going by this, predicting a probability of .011 when the actual observation label is 1 would result in a high loss value. In an ideal situation, a "perfect" model would have a log loss of 0. Looking at the loss function would make things even clearer -
+
+![Binary Cross Entropy Loss](/engineering-education/understanding-loss-functions-in-machine-learning/binary-cross-entropy-loss.PNG)
+
+*Source: [Rohan Varma](https://rohanvarma.me/Loss-Functions/)*
+
+Since binary classification means the classes take either 0 or 1, if y<sub>i</sub> = 0, that term ceases to exist and if y<sub>i</sub> = 1, the (1-y<sub>i</sub>) term becomes 0. Pretty clever, isn't it?
+
+#### Categorical Cross Entropy Loss
+Categorical Cross Entropy loss is essentially Binary Cross Entropy Loss expanded to multiple classes. One requirement when categorical cross entropy loss function is used is that the labels should be one-hot encoded. That way, there would be only one element in the target vector which will not be zero and hence, the elements which have zeros would be discarded by virtue of multiplication with zero. This property is extended to an activation function called softmax, more of which can be found in [this article](/engineering-education/activation-functions/)
```

Comment:
Give a single line explanation of what `one hot encoding` is.

Also, this line `That way, there would be only one element in the target vector which will not be zero and hence, the elements which have zeros would be discarded by virtue of multiplication with zero.` is a little unclear. If you could elaborate more on this or could give an example, it would be great.